src/compiler/msvc.rs

```diff
@@ -788,7 +788,20 @@ pub fn parse_arguments(
             );
         }
         Some(o) => {
-            if o.extension().is_none() && compilation {
+            if o.as_os_str()
```

Comment:
sorry for the latency.
compilation is gone from the check, is that expected?

Reply:
Np. It is intentional, at least. `compilation` is checked a bit earlier in this function (l760): this check was always true